timedctl/timedctl.py

```diff
@@ -25,708 +24,568 @@
-                # calculate duration
-                duration = attr["to-time"] - attr["from-time"]
-                # get task
-                task = activity_obj["relationships"]["task"]["data"]["id"]
-                # check if there is a report with the same comment that already exists
-                report = [
-                    x
-                    for x in reports
-                    if x["attributes"]["comment"] == attr["comment"]
-                    and x["relationships"]["task"]["data"]["id"] == task
-                ]
-                # if report has been found
-                if len(report) > 0:
-                    report = report[0]
-                    # deserialize the timedelta
-                    hours, minutes, seconds = report["attributes"]["duration"].split(
-                        ":",
-                    )
-                    old_duration = datetime.timedelta(
-                        hours=int(hours),
-                        minutes=int(minutes),
-                        seconds=int(seconds),
-                    )
-                    # calculate the new duration
-                    report["attributes"]["duration"] = old_duration + duration
-                    # update report
-                    timed.reports.patch(
-                        report["id"],
-                        report["attributes"],
-                        {"task": task},
-                    )
-                else:
-                    # create report
-                    res = timed.reports.post(
-                        {
-                            "duration": duration,
-                            "comment": attr["comment"],
-                        },
-                        {"task": task},
+
+        if res.status_code == requests.codes["created"]:
+            msg(f"Activity {comment} started successfully.")
+            return
+        # handle exception
+        error_handler("ERR_ACTIVITY_START_FAILED")
+
+    def stop_activity(self):
+        """Stop current activity."""
+        if not self.timed.activities.current:
+            error_handler("ERR_NO_CURRENT_ACTIVITY")
+        else:
+            self.timed.activities.stop()
+        msg("Activity stopped successfully.")
+
+    def show_activity(self, short):
+        """Show current activity."""
+        current_activity = self.timed.activities.get(
+            filters={"active": True},
+            include="task,task.project,task.project.customer",
+        )
+        if current_activity:
+            activity_obj = current_activity[0]
+            comment = " > " + activity_obj["attributes"]["comment"] if not short else ""
+            start = activity_obj["attributes"]["from-time"].strftime("%H:%M:%S")
+            msg(
+                f"Current activity: {self.format_activity(activity_obj)}{comment}"+
+                f"  (Since {start})",
+            )
+        else:
+            error_handler("ERR_NO_CURRENT_ACTIVITY")
```

Comment:
This can be refactored like this
```suggestion
        if not current_activity:
             error_handler("ERR_NO_CURRENT_ACTIVITY")
        activity_obj = current_activity[0]
        comment = " > " + activity_obj["attributes"]["comment"] if not short else ""
        start = activity_obj["attributes"]["from-time"].strftime("%H:%M:%S")
        msg(
            f"Current activity: {self.format_activity(activity_obj)}{comment}"+
            f"  (Since {start})",
        )
```